### DISCUSSION TITLE
Customize dropdown options in create/edit forms

### DISCUSSION THREAD

**Discussion by regisin** [aminalaee/sqladmin#641] [Q&A] [answered]
Hi, this might be supported already, but I couldn't figure out how to customize the <option> fields in the edit form when it reads relationship fields.

Currently, the options listed are displayed apparently by their python type and (memory) address:
![image](https://github.com/aminalaee/sqladmin/assets/794426/c226717e-3ce7-40d7-a223-7f4ebbb32122)

This screenshot is when adding/editing a user. Can I display `Plan.description` instead?

**Comment by aminalaee** [comment, marked as answer]
Yeah sure, just define `__str__` method for your models.
Feel free to add this to the docs if you find it useful.

**Reply by jik92** [reply]
thx very much!

**Reply by wnowicki** [reply]
I've found it very useful so I updated docs  https://github.com/aminalaee/sqladmin/pull/917

**Comment by danilo-nzyte** [comment]
@regisin can I ask for a code snippet of how your related field in your form is working?

I've tried to use as an example:

```
class CustomerProductsAdmin(ModelView, model=CustomerProducts):
    column_list = ["id", "name"]
    form_create_rules = ["name", "customer.name"]
```

Where I have a relationship defined between CustomerProducts and Customer (the relationship is called customer in the model)

**Reply by regisin** [reply]
I actually can't, that's a long gone project. Is the relationship working? I Remember not customizing at all the `column_list` and `form_create_rules`.

**Reply by danilo-nzyte** [reply]
Thanks for the reply, I literally just this second worked out the issue.

Out of interest - have you used this package successfully for a project at all or were you just testing things out?

The relationship fields were flagged as primary keys and therefore weren't displayed in the form create fields by default (`form_include_pk` is False by default. Changing this to True isn't the correct solution for me.

I've removed the definition of the primary key on the model (it was incorrect) and now it works as per your screenshot. Then I implemented the fix suggested by @aminalaee which is a final working solution.

Implementation below for anybody else reading.

`models.py`
```python
class CustomDefaultMixin(object):
    @declared_attr
    def __tablename__(cls):
        return cls.__name__

    id = Column(Integer, primary_key=True)
    is_deleted = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    modified_at = Column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

    def __str__(cls):
        return cls.name

class CustomerRateCard(Base, CustomDefaultMixin):
    name = Column(String)
    customer_id = Column(Integer, ForeignKey(Customer.id))

    customer = relationship("Customer", foreign_keys="CustomerRateCard.customer_id")
```

app.py
```python
class CustomerRateCardAdmin(AdminView, model=CustomerRateCard):
    # Nav settings
    name = "Customer Rate Card"
    name_plural = "Customer Rate Cards"
    icon = "fa-solid fa-user"

    # View settings
    column_list = ["id", "customer.name", "name", "customer_id", "created_at", "modified_at"]
    form_create_rules = ["customer", "name"]
```

<img width="1463" alt="image" src="https://github.com/user-attachments/assets/1b9a05bd-cdee-46f9-9d21-9779accfc277" />

Appreciate you getting back to me :)
